### PR TITLE
Problems API: simplify/deduplicate test code, increase coverage

### LINKED ISSUE #27352: Review and extend test coverage for Problems API
The Problems API surface vastly changed over the last few weeks. We should revisit the test coverage and make sure we have sufficient test coverage.

### PR DESCRIPTION
**Pull request by reinsch82**
Fixes #27352

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
I reduced a bit of duplication and fix a few tests to actually validate what is expected.
Didn't find a lot of missing coverage.

File: platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblems.java [hunk 1]
```diff
@@ -40,10 +40,6 @@ public DefaultProblems(ProblemEmitter emitter, List<ProblemTransformer> transfor
         internalReporter = createReporter(DefaultProblemCategory.GRADLE_CORE_NAMESPACE, emitter, transformers);
     }
 
-    public void setEmitter(ProblemEmitter emitter) {
-        this.emitter = emitter;
-    }
-
```
Comment (donat) [inline platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblems.java:46]: 👍

File: platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r86/ProblemProgressEventCrossVersionTest.groovy [hunk 2]
```diff
@@ -165,39 +157,29 @@ class ProblemProgressEventCrossVersionTest extends ToolingApiSpecification {
         ''                         | null            | ''                                          | null
     }
 
+    def runTask(){
```
Comment (donat) [inline platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r86/ProblemProgressEventCrossVersionTest.groovy:160]: 💅 There are a few cases in this PR where the formatting is slightly off. There's no space between the method closing parenthesis and the opening curly brace. A quick `Reformat code` action on the changed files would be useful.

```suggestion
    def runTask() {
```

**Review by donat — APPROVED** [review]

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle merge

**Pull request merged**